src/core/components/ChatView.tsx

```diff
@@ -7,6 +7,8 @@ import { useAppContext } from '../context/useAppContext';
 import { NavBar } from './NavBar';
 import type { ChatMessage } from '../stores/messageHistoryStore';
 import { useMessageHistory } from '../hooks/useMessageHistory';
+import { isWithinTokenLimit } from 'gpt-tokenizer';
```

Comment:
This is a very useful method that they expose that is ideal for the use case of disabling the input if a user attempts to input a ton of text that itself would overwhelm the context window.